### PR TITLE
[ADLS] integrate virtual network rule to account mgmt

### PR DESCRIPTION
**Pull request by lewu-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by lewu-msft** [comment]
@tjprescott Any detailed guidance?

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py [hunk 1]
```diff
@@ -105,6 +105,43 @@
     short-summary: Deletes a firewall rule in a Data Lake Store account.
 """
 
+helps['dls account virtual-network'] = """
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py]: This command group should be called `dls account network-rule`

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py [hunk 2]
```diff
@@ -105,6 +105,43 @@
...
+    type: group
+    short-summary: (PREVIEW) Manage Data Lake Store account virtual network rules.
+"""
+
+helps['dls account virtual-network create'] = """
+    type: command
+    short-summary: Creates a virtual network rule in a Data Lake Store account.
+    parameters:
+        - name: --subnet-id
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py]: This parameter should be called `--subnet` and accept the name or ID of the subnet in question.  You must also expose `--vnet-name` that is used to resolve the subnet when `--subnet` is provided as a name.
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py [hunk 3]
```diff
@@ -105,6 +105,43 @@
...
+          type: string
+          short-summary: 'The subnet id for the virtual network rule.'
+        - name: --virtual-network-rule-name
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py]: This should be aliased to `--name/-n` and that implies there's a good chance that `account_name` (which is likely aliased to --name) needs to be aliased to `--account-name` for this command group.

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/commands.py [hunk 4]
```diff
@@ -49,6 +55,14 @@ def load_command_table(self, _):
         g.command('show', 'get')
         g.command('delete', 'delete')
 
+    # account virtual network rule operations
+    with self.command_group('dls account virtual-network', dls_virtual_network_sdk, client_factory=cf_dls_account_virtual_network) as g:
+        g.custom_command('create', 'add_adls_virtual_network_rule')
+        g.command('update', 'update')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/commands.py]: This needs to be a generic update command. https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#generic-update-commands

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/commands.py [hunk 5]
```diff
@@ -49,6 +55,14 @@ def load_command_table(self, _):
...
+        g.command('list', 'list_by_account')
+        g.command('show', 'get')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/commands.py]: This needs to use `g.show_command` https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#standard-command-types

**Comment by williexu** [comment]
@lewu-msft ping, code-complete for our next release is tomorrow. If the changes are not addressed by then, this will probably be set for the subsequent release.

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py [hunk 6]
```diff
@@ -56,6 +59,25 @@ def load_arguments(self, _):
...
+    #     # c.argument('name', options_list=['--name', '-n'])
+    #
+    #     c.argument('virtual_network_subnet_id',
+    #                options_list=['--subnet'],
+    #                help='Name or ID of the subnet that allows access to ADLS. '
+    #                'If subnet name is provided, --vnet-name must be provided.')
+    #
+    #     c.argument('ignore_missing_vnet_service_endpoint',
+    #                options_list=['--ignore-missing-endpoint', '-i'],
+    #                help='Create firewall rule before the virtual network has vnet service endpoint enabled')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py]: If not being used, this should be removed.

**Comment by williexu** [comment]
@lewu-msft please bump the version for dla to make the CI pass

**Comment by lewu-msft** [comment]
stuck with those CI script checks.

File: pylintrc [hunk 7]
```diff
@@ -21,6 +21,8 @@ init-import=yes
 max-locals=25
 # Maximum number of branch for function / method body
 max-branches=20
+# Maximum number of statements in function / method body
+max-statements=80
```
Comment (tjprescott) [inline pylintrc]: Why is this here?

Comment (lewu-msft) [reply pylintrc]: to avoid pylint error like below
```
=== Begin testing
Run pylint with 2 proc.
Using config file /home/travis/build/Azure/azure-cli/pylintrc
************* Module azure.cli.command_modules.dls.commands
R: 19, 0: Too many statements (55/50) (too-many-statements)
------------------------------------
Your code has been rated at 10.00/10
The command "./scripts/ci/test_static.sh" exited with 8.
cache.2
store build cache
0.01s
134.83schange detected (content changed, file is created, or file is deleted):
/home/travis/.cache/pip/http/0/1/a/4/0/01a40e69133cae00a418944df93058fcdf9d6f64cf3656d5716568c0
/home/travis/.cache/pip/http/0/1/c/d/8/01cd8cc532ed9e2c3c5d2358dc9612d67ba9b890279a30d6a977235a
/home/travis/.cache/pip/http/0/4/e/0/1/04e017851dbfe81db92d092eaaf2526622c3950bcb8f2696e57143da
/home/travis/.cache/pip/http/0/d/f/a/c/0dfac344d300bdb90276ae93aedd26c65c7200784853af7dee358310
/home/travis/.cache/pip/http/1/2/0/9/7/12097cd615f100a315b54cf93aa5c9dfc7812252795327e660b78a60
/home/travis/.cache/pip/http/1/7/0/e/f/170ef7ade34dbc769581affb285d6c0f3b5f57bd93d4ea3f8eb0266f
/home/travis/.cache/pip/http/1/8/2/5/d/1825dc308e1c96bbfb6d6a06031849c56b2187de1c0e0decbd487984
/home/travis/.cache/pip/http/1/a/8/f/4/1a8f4c08b167173ed0f4c81934143a0f91d1155514fbfe15c8b21da1
/home/travis/.cache/pip/http/1/f/3/6/e/1f36e4fb24023fde97b032397872135c7514eeb07753ac9e6c41bfd9
/home/travis/.cache/pip/http/2/1/3/7/f/2137f603f620cd2c8b12df96008a50a6038fc764aaf0d243c22963d6
/home/travis/.cache/pip/http/2/5/5/b/8/25
...
changes detected, packing new archive
.
uploading archive
failed to upload cache
* Hostname was NOT found in DNS cache
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 209.85.234.128...
*   Trying 2607:f8b0:4001:c00::80...
* Immediate connect fail for 2607:f8b0:4001:c00::80: Cannot assign requested address
  0     0    0     0    0     0      0      0 --:--:--  0:02:06 --:--:--     0* connect to 209.85.234.128 port 443 failed: Connection timed out
* Failed to connect to travis-cache-production-org-gce.storage.googleapis.com port 443: Connection timed out
* Closing connection 0
curl: (7) Failed to connect to travis-cache-production-org-gce.storage.googleapis.com port 443: Connection timed out
Done. Your build exited with 1.
```

Comment (tjprescott) [reply pylintrc]: In that case, just put `# pylint: disable=too-many-statements` at the offending line of code.

**Comment by lewu-msft** [comment]
@tjprescott any suggestions where I can find the bad import? Thx

**Comment by tjprescott** [comment]
@lewu-msft set up a Skype meeting with me and we can check the code via screenshare. I didn't see anything in your PR that stuck out.

File: azure-cli2017.pyproj [hunk 8]
```diff
@@ -1330,6 +1329,15 @@
...
     </Interpreter>
+    <Interpreter Include="..\venv\">
+      <Id>venv</Id>
+      <Version>3.6</Version>
+      <Description>venv (Python 3.6 (32-bit))</Description>
+      <InterpreterPath>Scripts\python.exe</InterpreterPath>
+      <WindowsInterpreterPath>Scripts\pythonw.exe</WindowsInterpreterPath>
+      <PathEnvironmentVariable>PYTHONPATH</PathEnvironmentVariable>
+      <Architecture>X86</Architecture>
+    </Interpreter>
```
Comment (tjprescott) [inline azure-cli2017.pyproj]: Revert the changes to this file

File: azure-cli20171.pyperf [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
+<ProfilingTarget xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <ProjectTarget>
+    <TargetProject>938454f7-93bd-41a7-84b2-3c89d64b969d</TargetProject>
+    <FriendlyName>azure-cli2017</FriendlyName>
+  </ProjectTarget>
+</ProfilingTarget>
```
Comment (tjprescott) [inline azure-cli20171.pyperf]: Remove this file

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/tests/latest/test_dls_commands.py [hunk 10]
```diff
@@ -375,16 +375,14 @@ def test_dls_account_mgmt(self, resource_group):
             'subnet_id': '/subscriptions/9e1f0ab2-2f85-49de-9677-9da6f829b914/resourceGroups/lewu-rg/providers/Microsoft.Network/virtualNetworks/lewuVNET/subnets/default',
             'updated_subnet_id': '/subscriptions/9e1f0ab2-2f85-49de-9677-9da6f829b914/resourceGroups/lewu-rg/providers/Microsoft.Network/virtualNetworks/lewuVNET/subnets/updatedSubnetId'
         })
-        self.cmd('dls account network-rule create -g {rg} -n {dls} --vnet-name {vnet} --subnet {subnet_id}')
-        self.cmd('dls account network-rule show -g {rg} -n {dls} --vnet-name {vnet}', checks=[
+        self.cmd('dls account network-rule create -g {rg} -n {dls} --virtual-network-rule-name {vnet} --subnet {subnet_id}')
+        self.cmd('dls account network-rule show -g {rg} -n {dls} --virtual-network-rule-name {vnet}', checks=[
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/tests/latest/test_dls_commands.py]: Here, `virtual_network_rule_name` should be aliased to `--name/-n` not DLS.

Comment (lewu-msft) [reply src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/tests/latest/test_dls_commands.py]: Thanks. should I change from ```--virtual-network-rule-name``` to ```--name``` ?  What do you mean by ```not DLS```

Comment (tjprescott) [reply src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/tests/latest/test_dls_commands.py]: Your example shows `-n {dls}` but name should refer to the rule name in this case, not the DLS account instance.

**Comment by lewu-msft** [comment]
after bumping ADLA dependency version, can't pass CI check now. will make the changes.

**Comment by troydai** [comment]
@lewu-msft the code complete date for this sprint is 8/22. Anything I can help with this PR?

File: azure-cli20171.pyperf (hunk 9 above)
Comment (tjprescott) [reply azure-cli20171.pyperf]: Still needs to be removed.

**Comment by lewu-msft** [comment]
@troydai The only thing failed is to update the recording of DLA, I still have errors when running DLA tests

**Comment by troydai** [comment]
@lewu-msft cool. let me know if there is anything i can help.

**Comment by lewu-msft** [comment]
I reverted the changes about DLA dependency upgrade. Will open a new PR for that after merging this one. :)

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/tests/latest/test_dls_commands.py [hunk 11]
```diff
@@ -366,6 +368,32 @@ def test_dls_account_mgmt(self, resource_group):
...
         ])
+
+        # test virtual network rule crud
+        self.kwargs.update({
+            'vnet': 'lewuVNET',
+            'subnet_id': '/subscriptions/9e1f0ab2-2f85-49de-9677-9da6f829b914/resourceGroups/lewu-rg/providers/Microsoft.Network/virtualNetworks/lewuVNET/subnets/default',
+            'updated_subnet_id': '/subscriptions/9e1f0ab2-2f85-49de-9677-9da6f829b914/resourceGroups/lewu-rg/providers/Microsoft.Network/virtualNetworks/lewuVNET/subnets/updatedSubnetId'
+        })
+        self.cmd('dls account network-rule create -g {rg} --account {dls} --virtual-network-rule-name {vnet} --subnet {subnet_id}')
+        self.cmd('dls account network-rule show -g {rg} --account {dls} --virtual-network-rule-name {vnet}', checks=[
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/tests/latest/test_dls_commands.py]: `--virtual-network-rule-name` should be aliased to `--name/-n` for all commands within the `dls account network-rule` group.

**Comment by troydai** [comment]
@tjprescott ?

**Comment by tjprescott** [comment]
@lewu-msft let's do a quick in-person or Skype review to make sure everything is good-to-go

**Comment by lewu-msft** [comment]
@tjprescott  more suggestions?
```

(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule -h

Group
    az dls account network-rule : (PREVIEW) Manage Data Lake Store account virtual network rules.

Commands:
    create : Creates a virtual network rule in a Data Lake Store account.
    delete : Deletes a virtual network rule in a Data Lake Store account.
    list   : Lists virtual network rules in a Data Lake Store account.
    show   : Get the details of a virtual network rule in a Data Lake Store account.
    update : Updates a virtual network rule in a Data Lake Store account.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule create -h

Command
    az dls account network-rule create : Creates a virtual network rule in a Data Lake Store
    account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --name -n      [Required] : The virtual network rule name.
    --subnet       [Required] : The subnet name or id for the virtual network rule.
    --resource-group -g       : If not specified, will attempt to discover the resource group for
                                the specified Data Lake Store account.
    --vnet-name               : The name of the virtual network rule.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule delete -h

Command
    az dls account network-rule delete : Deletes a virtual network rule in a Data Lake Store
    account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --resource-group -g       : If not specified, will attempt to discover the resource group for
                                the specified Data Lake Store account.

Resource Id Arguments
    --ids                     : One or more resource IDs (space-delimited). If provided, no other
                                'Resource Id' arguments should be specified.
    --name -n                 : The virtual network rule name.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule list -h

Command
    az dls account network-rule list : Lists virtual network rules in a Data Lake Store account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --resource-group -g       : If not specified, will attempt to discover the resource group for
                                the specified Data Lake Store account.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule show -h

Command
    az dls account network-rule show : Get the details of a virtual network rule in a Data Lake
    Store account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --resource-group -g       : If not specified, will attempt to discover the resource group for
                                the specified Data Lake Store account.

Resource Id Arguments
    --ids                     : One or more resource IDs (space-delimited). If provided, no other
                                'Resource Id' arguments should be specified.
    --name -n                 : The virtual network rule name.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule update -h

Command
    az dls account network-rule update : Updates a virtual network rule in a Data Lake Store
    account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --subnet       [Required] : Name or ID of the subnet that allows access to DLS. If subnet name
                                is provided, --name must be provided.
    --resource-group -g       : If not specified, will attempt to discover the resource group for
                                the specified Data Lake Store account.
    --vnet-name               : The virtual network rule name.

Generic Update Arguments
    --add                     : Add an object to a list of objects by specifying a path and key
                                value pairs.  Example: --add property.listProperty <key=value,
                                string or JSON string>.
    --force-string            : When using 'set' or 'add', preserve string literals instead of
                                attempting to convert to JSON.
    --remove                  : Remove a property or an element from a list.  Example: --remove
                                property.list <indexToRemove> OR --remove propertyToRemove.
    --set                     : Update an object by specifying a property path and value to set.
                                Example: --set property1.property2=<value>.

Resource Id Arguments
    --ids                     : One or more resource IDs (space-delimited). If provided, no other
                                'Resource Id' arguments should be specified.
    --name -n                 : The virtual network rule name.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.
```

**Comment by troydai** [comment]
None of the `--resource-group` are in the `Resource Id Arguments` group. Why?

**Comment by troydai** [comment]
Please read: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-the-ids-parameter

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py [hunk 12]
```diff
@@ -64,6 +64,7 @@ def load_arguments(self, _):
     #####
     with self.argument_context('dls account network-rule') as c:
         c.argument('account_name', datalake_store_name_type, options_list=['--account-name'], id_part=None)
+        c.argument('resource_group_name', resource_group_name_type, id_part='name')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py]: id_part for resource group name should definitely not be name. It should only be `resource_group` (at least in this context)

Comment (lewu-msft) [reply src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py]: Thanks, changed.
```
(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule delete -h

Command
    az dls account network-rule delete : Deletes a virtual network rule in a Data Lake Store
    account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.

Resource Id Arguments
    --ids                     : One or more resource IDs (space-delimited). If provided, no other
                                'Resource Id' arguments should be specified.
    --name -n                 : The virtual network rule name.
    --resource-group -g       : Name of resource group. You can configure the default group using
                                `az configure --defaults group=<name>`.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule show -h

Command
    az dls account network-rule show : Get the details of a virtual network rule in a Data Lake
    Store account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.

Resource Id Arguments
    --ids                     : One or more resource IDs (space-delimited). If provided, no other
                                'Resource Id' arguments should be specified.
    --name -n                 : The virtual network rule name.
    --resource-group -g       : Name of resource group. You can configure the default group using
                                `az configure --defaults group=<name>`.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule update -h

Command
    az dls account network-rule update : Updates a virtual network rule in a Data Lake Store
    account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --subnet       [Required] : Name or ID of the subnet that allows access to DLS. If subnet name
                                is provided, --name must be provided.
    --vnet-name               : The virtual network rule name.

Generic Update Arguments
    --add                     : Add an object to a list of objects by specifying a path and key
                                value pairs.  Example: --add property.listProperty <key=value,
                                string or JSON string>.
    --force-string            : When using 'set' or 'add', preserve string literals instead of
                                attempting to convert to JSON.
    --remove                  : Remove a property or an element from a list.  Example: --remove
                                property.list <indexToRemove> OR --remove propertyToRemove.
    --set                     : Update an object by specifying a property path and value to set.
                                Example: --set property1.property2=<value>.

Resource Id Arguments
    --ids                     : One or more resource IDs (space-delimited). If provided, no other
                                'Resource Id' arguments should be specified.
    --name -n                 : The virtual network rule name.
    --resource-group -g       : Name of resource group. You can configure the default group using
                                `az configure --defaults group=<name>`.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule create -h

Command
    az dls account network-rule create : Creates a virtual network rule in a Data Lake Store
    account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --name -n      [Required] : The virtual network rule name.
    --subnet       [Required] : The subnet name or id for the virtual network rule.
    --resource-group -g       : Name of resource group. You can configure the default group using
                                `az configure --defaults group=<name>`.
    --vnet-name               : The name of the virtual network rule.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.


(venv) E:\Users\lewu\workspace\lewu-msft\azure-cli>az dls account network-rule list -h

Command
    az dls account network-rule list : Lists virtual network rules in a Data Lake Store account.

Arguments
    --account-name [Required] : Name of the Data Lake Store account.
    --resource-group -g       : Name of resource group. You can configure the default group using
                                `az configure --defaults group=<name>`.

Global Arguments
    --debug                   : Increase logging verbosity to show all debug logs.
    --help -h                 : Show this help message and exit.
    --output -o               : Output format.  Allowed values: json, jsonc, table, tsv, yaml.
                                Default: json.
    --query                   : JMESPath query string. See http://jmespath.org/ for more information
                                and examples.
    --subscription            : Name or ID of subscription. You can configure the default
                                subscription using `az account set -s NAME_OR_ID`.
    --verbose                 : Increase logging verbosity. Use --debug for full debug logs.
```

**Comment by troydai** [comment]
@lewu-msft when will you start the work to update 'azure-mgmt-datalake-analytics==0.2.0', to `0.5.0`?

**Comment by lewu-msft** [comment]
> @lewu-msft when will you start the work to update 'azure-mgmt-datalake-analytics==0.2.0', to `0.5.0`?

@troydai Thanks! Will open another PR to update ADLA soon
